compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/desugar/Desugar.java

```diff
@@ -760,7 +780,6 @@ public void visit(BLangPackage pkgNode) {
         pkgNode.typeDefinitions = rewrite(pkgNode.typeDefinitions, env);
         pkgNode.xmlnsList = rewrite(pkgNode.xmlnsList, env);
         pkgNode.constants = rewrite(pkgNode.constants, env);
-        pkgNode.globalVars = rewrite(pkgNode.globalVars, env);
```

Comment:
With this we are changing the order in which different constructs are desugared, right? Hope this wouldn't cause issues in `serviceDesugar`, `annotationDesugar`, etc. 

For example, in the annotation desugar we populate a global map variable. Wonder if it still gets rewritten after this change. Can you try several different scenarios and validate?

Reply:
We can move global var desugar logic to here if so. Should we move?